spec/treasury_and_analytics.md

```diff
@@ -0,0 +1,135 @@
+# Treasury and Analytics
+
+This specification provides details of BitDAO treasury and analytics pages, as well as associated
+required services.
+
+We first describe how services are constructed, then we describe how the data is presented to users.
+
+The treasury and analytics approach taken here is based on daily values. This specification is not
+intended to serve shorter periods or live updates (the backend services will need to be re-evaluated
+for performance and flexibility, and likely requires rebuilding).
+
+## Features
+
+- Idempotent pipeline
+- Static page generation
+- Serverless using object storage, i.e., no key-value stores or relational database administration
+- Decentralized analytics by enabling self-contained pages
+
+## Analytics
+
+Data retrieval:
+
+- A serverless code executable retrieves token quantities and spot prices once a day.
+- Retrieved values are saved to S3 (as CSV or JSON) using the following filename convention:
+    - {token}-{yyyy}-{mm}-{dd}.json- If spot prices are pulled from multiple sources, use the
+      following filename convention:
+- Optional: Retrieve spot prices from multiple sources and store in the same file
+
+#### Token quantity specification:
+
+| Field | Type | Description |
+| ---- | ---- | ---- |
+| yyyy | number | Year of retrieved value: MUST equal 2021 | 
+| mm | number | Month of retrieved value: MUST be between 1 and 12 | 
+| dd | number | Day of retrieved value: MUST be between 1 and 31 (validation tbd) |
+| token | String | Short token identifier, e.g., BIT, BTC, ETH, USDC, USDT | 
+| quantity | BigInt | Number of tokens: MUST be greater than or equal to 0: MUST support BigInt |
+| source | String | The url where this row's value was obtained, e.g., https://coinmarketcap.com/currencies/bitdao/ | 
+
+#### Spot price specification:
+
+| Field | Type | Description |
+| ---- | ---- | ---- |
+| yyyy | number | Year of retrieved value: MUST equal 2021 |
```

Comment:
question: isn't validation of the year to 2021 going to break the system next year?

Reply:
yes, so these constraint need to be adjustable in the new year, or automated. Or even removed?

Reply:
Having the date constrained to a valid calendar date makes sense. Could even add a minimum date?

Reply:
I agree